extensions/sql_db_interfaces.go

```diff
@@ -55,9 +55,11 @@ type transactionalCRUD interface {
 	UpdateProcessExecution(ctx context.Context, row ProcessExecutionRowForUpdate) error
 
 	SelectAsyncStateExecutionForUpdate(ctx context.Context, filter AsyncStateExecutionSelectFilter) (*AsyncStateExecutionRow, error)
+	SelectRunningAsyncStateExecutionCount(ctx context.Context, processExecutionId uuid.UUID) (int, error)
```

Comment:
I forgot to mention it last PR, it’s usually easier to test the code, if you adding/change a test case in postgres_test.go to just verify the database code path is working as expected. Based on that, testing other engine and service layer would be much easier